src/main/java/com/machowina/service/TicketServiceImp.java

```diff
@@ -0,0 +1,181 @@
+package com.machowina.service;
+
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.List;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.machowina.exception.EntityNotFoundException;
+import com.machowina.exception.TicketAlreadyRunning;
+import com.machowina.exception.TicketAlreadyStopped;
+import com.machowina.exception.TicketNotStoppedException;
+import com.machowina.model.Car;
+import com.machowina.model.ParkingRates;
+import com.machowina.model.ParkingTicket;
+import com.machowina.model.ParkingZone;
+import com.machowina.model.User;
+import com.machowina.repository.TicketRepository;
+
+@Service
+public class TicketServiceImp implements TicketService {
+	
+	private final CarService carService;
+	private final ParkingZoneService zoneService;
+	private final ParkingRatesService parkingRatesService;
+	
+	private final TicketRepository ticketRepository;
+	
+
+	public TicketServiceImp(CarService carService, ParkingZoneService zoneService,
+			TicketRepository ticketRepository, ParkingRatesService parkingRatesService) {
+		super();
+		this.carService = carService;
+		this.zoneService = zoneService;
+		this.ticketRepository = ticketRepository;
+		this.parkingRatesService = parkingRatesService;
+	}
+
+	@Override
+	@Transactional
```

Comment:
co to daje? czy to wystarczy?

Reply:
O ile rozumiem przy wystąpieniu błędu podczas wykonywania transakcji nastąpi cofnięcie wszystkiego do stanu początkowego. Ale moja wiedza o transakcjach jest raczej podstawowa.